### PR TITLE
Adds the Ancient Beret Altar

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

Adds a beret altar, found in the depths of maintenance. It creates berets.
Video: https://streamable.com/bwajfe
Alternative to #54255 (not anymore :((( )
sprites from /vg/station

## Why It's Good For The Game

makes some fun berets and adds cool random fluff to maintenance

## Changelog
:cl: Fikou, Floyd
add: In the depths of maintenance, you can now find an ancient cult altar...
/:cl:

### CODE REVIEW

**Comment by EgorDinamit** [comment]
Praised be random.

**Comment by itseasytosee** [comment]
image
hon hon

**Comment by ArcaneMusic** [comment]
If nothing else, I'm all about that cult tgui theme.

**Comment by ATH1909** [comment]
oh my god

y e s

**Comment by ATH1909** [comment]
also, the logs from that demonstration video seem to imply that you had initially used a wraith as the mob guarding the altar, but you died to it, so you replaced it with an artificer in your next run through...

**Comment by MimicFaux** [comment]
regular cultists should be allowed to make these in their pursuits of freeing Nar'sie

**Comment by KathyRyals** [comment]
This has NO right to be as good as it is. My God.

**Review by Twaticus — APPROVED** [review]
youre insane

File: code/game/objects/structures/maintenance.dm [hunk 1]
```diff
@@ -99,3 +105,112 @@ at the cost of risking a vicious bite.**/
...
+	name = "strange structure"
+	desc = "What is this? Who put it on this station? And why does it eminate <span class='hypnophrase'>beret energy?</span>"
+	icon_state = "altar"
+	cultist_examine_message = "Even you don't understand the eldritch magic behind this."
+	break_message = "<span class='warning'>The structure shatters, leaving only a demonic screech!</span>"
+	break_sound = 'sound/magic/demon_dies.ogg'
+	light_color = LIGHT_COLOR_BLOOD_MAGIC
+	light_range = 2
+	var/beret_color = "#ffffff"
+	var/status = ALTAR_INACTIVE
```
Comment (Rohesie) [inline code/game/objects/structures/maintenance.dm:119]: Document the variables and use color defines, please.

File: code/game/objects/structures/maintenance.dm [hunk 2]
```diff
@@ -99,3 +105,112 @@ at the cost of risking a vicious bite.**/
...
+
+/obj/structure/destructible/cult/beret_altar/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/melee/cultblade/dagger) && iscultist(user) && status)
+		to_chat(user, "<span class='notice'>[src] is creating something, you can't move it!</span>")
+	else
+		return ..()
```
Comment (Rohesie) [inline code/game/objects/structures/maintenance.dm:125]:
```suggestion
		return
	return ..()
```

File: code/game/objects/structures/maintenance.dm [hunk 3]
```diff
@@ -99,3 +105,112 @@ at the cost of risking a vicious bite.**/
...
+
+/obj/structure/destructible/cult/beret_altar/proc/beret_create()
+	status = ALTAR_INACTIVE
+	update_icon()
+	visible_message("<span class='warning'>[src] eminates a flash of light and creates... a beret?</span>")
+	for(var/mob/living/mob in viewers(7, src))
+		mob.flash_act()
+	var/obj/item/clothing/head/beret/greyscale/beret = new /obj/item/clothing/head/beret/greyscale(get_turf(src))
+	beret.add_atom_colour(beret_color, ADMIN_COLOUR_PRIORITY)
+	cooldowntime = world.time + 1 MINUTES
```
Comment (Rohesie) [inline code/game/objects/structures/maintenance.dm:176]: Can you use the macros of `cooldowns.dm`?

**Comment by Rohesie** [comment]
Will this be usable by cultists? Same concern on the produced runed metal.

**Comment by wesoda25** [comment]
This PR doesn't make sense if it isn't an alternative

**Comment by Fikou** [comment]
> Will this be usable by cultists? Same concern on the produced runed metal.

why wouldnt it be usable by cultists, its usable by anyone
runed metal is useless for anything except creating golems as a crewmember, and 1 runed metal isnt gonna give you anything as a cultist, you can get guarranteed runed metal from curator's office but nobody does that, also this lootsite is really rare

**Comment by tralezab** [comment]
Roundstart magic boo, boooo! Are we all just lambs to the thematic slaughter?

**Comment by LemonInTheDark** [comment]
This uh, this pr feels like a meme, but I'm not sure that matters

**Comment by Rohesie** [comment]
The feature is nice, but I'm not sure maintenance is the right place. How about a space ruin?

**Comment by KathyRyals** [comment]
No more space content please, I beg..

**Comment by Fikou** [comment]
> The feature is nice, but I'm not sure maintenance is the right place. How about a space ruin?

we have an offstation content freeze, and this is a cult structure, in maintenance, where cult hides, we already have cult objects in maintenance (runes can spawn rarely)

**Comment by Rohesie** [comment]
> 
> 
> > The feature is nice, but I'm not sure maintenance is the right place. How about a space ruin?
> 
> we have an offstation content freeze, and this is a cult structure, in maintenance, where cult hides, we already have cult objects in maintenance (runes can spawn rarely)

Talk with oranges and see if they allow this off-station. I feel that for a maint feature it's a bit excessive, as arm commented.

**Comment by Fikou** [comment]
i dont feel its excessive, we already have literal runes and magic dies that can turn people into wizards, this is just a goofy harmless object like hobo seats

**Comment by necromanceranne** [comment]
Seems like a fun space ruin idea honestly...if it's transportable. Or, like, lets you replicate it somehow.

**Review by Rohesie — CHANGES_REQUESTED** [review]
Pending being moved away from maintenance.

**Comment by optimumtact** [comment]
not funny, magic and there's a ban on offstation content 

:-1: I see no reason to add this

**Comment by Pumpkinoe** [comment]
but oranges it's on station

**Comment by Rohesie** [comment]
I was suggesting adding it to offstation, because it definitely doesn't fit on maint.
Sorry, Fikou, looks like this magic little thing ain't happening.

**Pull request closed without merging**